src/details/ArborX_Ray.hpp

```diff
@@ -72,8 +115,8 @@ struct Ray
   KOKKOS_FUNCTION
   constexpr Vector const &direction() const { return _direction; }
 
-  Point _origin = {};
-  Vector _direction = {{0.f, 0.f, 0.f}};
+  // FIXME avoid breaking Wenjun's code
+  using Vector [[deprecated]] = ArborX::Experimental::Vector;
```

Comment:
This broke my builds (does not matter if it's in the current form or in `using Vector [[deprecated] = Vector;`) using gcc-7.5.0:
```
../src/details/ArborX_Ray.hpp:119:39: error: declaration of ‘using Vector = struct ArborX::Experimental::Vector’ [-fpermissive]
   using Vector [[deprecated]] = Vector;
                                       ^
../src/details/ArborX_Ray.hpp:30:8: error: changes meaning of ‘Vector’ from ‘struct ArborX::Experimental::Vector’ [-fpermissive]
 struct Vector : private Point
```
I can't a way to fix it, so unless someone has a better idea, may need to revert the PR.

Reply:
Try referring to Vector through an alias in the impl

Reply:
@dalg24 Don't understand you. I tried moving `using` to the top of the struct. I'm getting deprecation warnings in this case, so it's using alias, but I still get the error.

Reply:
Then, go with the full name to avoid the deprecation warnings, i.e. `ArborX::Experiemntal::Vector` instead of `Vector`.